pkg/apiserver/diagnose/compare.go

```diff
@@ -0,0 +1,491 @@
+		subRow1, ok := rowsMap1[label]
+		if ok {
+			continue
+		}
+		ratio, err := calculateDiffRatio(subRow1, subRow2, table)
+		if err != nil {
+			return nil, errors.Errorf("category %v,table %v, calculate diff ratio error: %v,  %v,%v", strings.Join(table.Category, ","), table.Title, err.Error(), subRow1, subRow2)
+		}
+
+		subJoinRows = append(subJoinRows, &newJoinRow{
+			row1:  subRow1,
+			row2:  subRow2,
+			ratio: ratio,
+		})
+	}
+
+	sort.Slice(subJoinRows, func(i, j int) bool {
+		return subJoinRows[i].ratio > subJoinRows[j].ratio
+	})
+	totalRatio := float64(0)
+	resultSubRows := make([][]string, 0, len(row1.SubValues))
+	for _, r := range subJoinRows {
+		totalRatio += r.ratio
+		resultSubRows = append(resultSubRows, r.genNewRow(table))
+	}
+
+	// row join with null row
+	if len(subJoinRows) == 0 {
+		if len(row1.Values) != len(row2.Values) {
+			totalRatio = 1
+		} else {
+			totalRatio, err = calculateDiffRatio(row1.Values, row2.Values, table)
+			if err != nil {
+				return nil, errors.Errorf("category %v,table %v, calculate diff ratio error: %v,  %v,%v", strings.Join(table.Category, ","), table.Title, err.Error(), row1.Values, row2.Values)
+			}
+		}
+	}
+
+	resultJoinRow := newJoinRow{
+		row1:  row1.Values,
+		row2:  row2.Values,
+		ratio: totalRatio,
+	}
+
+	resultRow := &TableRowDef{
+		Values:    resultJoinRow.genNewRow(table),
+		SubValues: resultSubRows,
+		ratio:     totalRatio,
+		Comment:   "",
+	}
+	return resultRow, nil
+}
+
+type newJoinRow struct {
+	row1  []string
+	row2  []string
+	ratio float64
+}
+
+func (r *newJoinRow) genNewRow(table *TableDef) []string {
+	newRow := make([]string, 0, len(r.row1)+len(r.row2))
+	ratio := convertFloatToString(r.ratio)
+	if len(r.row1) == 0 {
+		newRow = append(newRow, make([]string, len(r.row2))...)
+		newRow = append(newRow, ratio)
+		for i := range r.row2 {
+			if checkIn(i, table.joinColumns) {
+				newRow[i] = r.row2[i]
+			} else {
+				newRow = append(newRow, r.row2[i])
+			}
+		}
+		return newRow
+	}
+
+	newRow = append(newRow, r.row1...)
+	newRow = append(newRow, ratio)
+	if len(r.row2) == 0 {
+		newRow = append(newRow, make([]string, len(r.row1)-len(table.joinColumns))...)
+		return newRow
+	}
+	for i := range r.row2 {
+		if !checkIn(i, table.joinColumns) {
+			newRow = append(newRow, r.row2[i])
+		}
+	}
+	return newRow
+}
+
+func calculateDiffRatio(row1, row2 []string, table *TableDef) (float64, error) {
+	if len(table.compareColumns) == 0 {
+		return 0, nil
+	}
+	if len(row1) == 0 && len(row2) == 0 {
+		return 0, nil
+	}
+	if len(row1) == 0 || len(row2) == 0 {
+		return float64(1), nil
+	}
+	maxRatio := float64(0)
+	for _, idx := range table.compareColumns {
+		f1, err := parseFloat(row1[idx])
+		if err != nil {
+			return 0, err
+		}
+		f2, err := parseFloat(row2[idx])
+		if err != nil {
+			return 0, err
+		}
+		if f1 == f2 {
+			continue
+		}
+		ratio := (f2 - f1) / math.Max(f1, f2)
+		if math.Abs(ratio) > math.Abs(maxRatio) {
+			maxRatio = ratio
```

Comment:
How about add the max diff name in  `maxRatio` .
For example , if the max diff is avg(value), then join it to maxRatio as one string.

Reply:
![image](https://user-images.githubusercontent.com/26020263/75848795-1f935300-5e1e-11ea-943b-648be28d9c57.png)

After trying, I prefer the original format...